docs/notebooks/active_learning_for_binary_classification.pct.py

```diff
@@ -0,0 +1,187 @@
+# %% [markdown]
+# # Active Learning for Gaussian Process Classification Model
+
+# %%
+import gpflow
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+
+import trieste
+from trieste.acquisition.function import BayesianActiveLearningByDisagreement
+from trieste.acquisition.rule import OBJECTIVE
+from trieste.models.gpflow.models import VariationalGaussianProcess
+from trieste.objectives.utils import mk_observer
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+# %% [markdown]
+# ## The problem
+
+# %% [markdown]
+# In Trieste, it is also possible to query most interesting points for learning the problem, i.e we want to have as little data as possible to construct the best possible model (active learning). In this tutorial we will try to do active learning for binary classification problem using Bayesain Active Learning by Disagreement (BALD) for a Gaussian Process Classification Model.
+#
+# Our problem is classification problem of circle dataset. In this toturial we assume that the input space is continous so we can use continuous optimiser for our BALD acquisition function.
+
+# %%
+search_space = trieste.space.Box([-1, -1], [1, 1])
+input_dim = 2
+
+
+def circle(x):
+    return tf.cast((tf.reduce_sum(tf.square(x), axis=1, keepdims=True) - 0.5) > 0, tf.float64)
+
+
+# %%
+density = 100
+query_points = np.linspace(search_space.lower[0], search_space.upper[0], density)
+
+grid_query_points = np.meshgrid(*[query_points] * input_dim)
+query_points = np.vstack([g.ravel() for g in grid_query_points]).T
+observations = circle(query_points).numpy()
+
+plt.figure(figsize=(5, 5))
+plt.contour(*grid_query_points, np.reshape(observations, [density] * input_dim), levels=[0.5])
+idx = np.squeeze(observations).astype(bool)
+plt.scatter(query_points[idx][:, 0], query_points[idx][:, 1], label="1")
+plt.scatter(
+    query_points[np.logical_not(idx)][:, 0], query_points[np.logical_not(idx)][:, 1], label="0"
+)
+plt.legend()
+plt.show()
+
+# %% [markdown]
+# Let's generate some data for our initial model. Here we randomly sample 10 data points.
+
+# %%
+num_initial_points = 10
+X = search_space.sample(num_initial_points)
+observer = mk_observer(circle)
+initial_data = observer(X)
+
+# %% [markdown]
+# ## Modelling the binary classification task
+
+# %% [markdown]
+# For the binary classification model, we use the Variational Gaussian Process with Bernoulli likelihood. For more detail of this model, see <cite data-cite="Nickisch08a">[Nickisch et al.](https://www.jmlr.org/papers/volume9/nickisch08a/nickisch08a.pdf)</cite>.
+
+# %%
+from trieste.models.gpflow import VariationalGaussianProcess
+from trieste.models.optimizer import Optimizer
+
+optimizer = Optimizer(
+    optimizer=gpflow.optimizers.Scipy(), minimize_args={"options": dict(maxiter=100)}
+)
+
+
+def create_bo_model(data):
+    kernel = gpflow.kernels.SquaredExponential(lengthscales=[0.2, 0.2])
```

Comment:
we now have convenient gpflow model builders in `models/gpflow/builders.py`, you can use those to set up a VGP model with sensible defaults (there is also SVGP classification model)